File: jfingerprintmanager/src/test/java/com/jesusm/jfingerprintmanager/BaseTest.java
package com.jesusm.jfingerprintmanager;

import android.annotation.SuppressLint;
import android.content.Context;
import android.support.v4.app.FragmentManager;

import com.jesusm.jfingerprintmanager.base.FingerprintAssetsManager;
import com.jesusm.jfingerprintmanager.base.hardware.FingerprintHardware;
import com.jesusm.jfingerprintmanager.base.keystore.KeyStoreManager;
import com.jesusm.jfingerprintmanager.base.ui.FingerprintBaseDialogFragment;
import com.jesusm.jfingerprintmanager.base.ui.System;
import com.jesusm.jfingerprintmanager.encryption.Encoder;

import org.mockito.Mock;

public class BaseTest {
    protected static final String KEY_STORE_ALIAS = "fake_key";

    @Mock
    protected FingerprintHardware mockFingerprintHardware;
    @Mock
    protected KeyStoreManager mockKeyStoreManager;
    @Mock
    protected FragmentManager mockFragmentManager;
    @Mock
    protected FakeSystem mockSystem;
    @Mock
    private Context mockContext;

    @SuppressLint("VisibleForTests")
    protected JFingerprintManager createFingerPrintManager() {
        FingerprintAssetsManager fingerprintAssetsManager = new FingerprintAssetsManager(mockContext,
                mockFingerprintHardware, mockKeyStoreManager, KEY_STORE_ALIAS);
        return new JFingerprintManager(mockSystem, fingerprintAssetsManager, new FakeEncoder());
    }

    protected class FakeSystem implements System {

        @Override
        public void showDialog() { }

        @Override
        public void addDialogInfo(FingerprintBaseDialogFragment.Builder builder, FragmentManager fragmentManager) { }
    }

    public class FakeEncoder implements Encoder {

        @Override
        public String encode(byte[] message) {
            return "";
        }

        @Override
        public byte[] decode(String messageToDecode) {
            return new byte[0];
        }
    }
}

File: jfingerprintmanager/src/test/kotlin/com.jesusm.jfingerprintmanager/BaseTest.kt
package com.jesusm.jfingerprintmanager

import android.annotation.SuppressLint
import android.content.Context
import android.support.v4.app.FragmentManager
import com.jesusm.jfingerprintmanager.base.FingerprintAssetsManager
import com.jesusm.jfingerprintmanager.base.hardware.FingerprintHardware
import com.jesusm.jfingerprintmanager.base.keystore.KeyStoreManager
import com.jesusm.jfingerprintmanager.base.ui.FingerprintBaseDialogFragment
import com.jesusm.jfingerprintmanager.base.ui.System
import com.jesusm.jfingerprintmanager.encryption.Encoder
import com.nhaarman.mockito_kotlin.mock

open class BaseTest {
    companion object {
        val KEY_STORE_ALIAS: String = "fake_key"
    }

    val mockFingerprintHardware: FingerprintHardware = mock()
    val mockKeyStoreManager: KeyStoreManager = mock()
    val mockFragmentManager: FragmentManager = mock()
    val mockSystem: FakeSystem = mock()
    val mockContext: Context = mock()

    @SuppressLint("VisibleForTests")
    fun createFingerPrintManager(): JFingerprintManager {
        val fingerprintAssetsManager = FingerprintAssetsManager(mockContext,
                KEY_STORE_ALIAS, mockFingerprintHardware, mockKeyStoreManager)
        return JFingerprintManager(mockContext, KEY_STORE_ALIAS, mockSystem, fingerprintAssetsManager, FakeEncoder())
    }

    inner class FakeSystem : System {
        override fun showDialog() {

        }

        override fun addDialogInfo(builder: FingerprintBaseDialogFragment.Builder<out FingerprintBaseDialogFragment<*>, *>?, fragmentManager: FragmentManager?) {
        }
    }

    inner class FakeEncoder : Encoder {

        override fun encode(messageToEncode: ByteArray): String {
            return ""
        }

        override fun decode(messageToDecode: String): ByteArray {
            return ByteArray(0)
        }
    }
}